iterators_task_1/student/include/impl.h

```diff
@@ -10,4 +12,28 @@
  *
  * @warning YOU MUST USE STRING STREAMS AND STREAM ITERATORS!
  */
-std::string reverseWords(const std::string& input);
+std::string reverseWords(const std::string& input)
+{
+    std::vector<std::string> token;
+    std::istringstream in(input);
+    std::ostringstream out;
+    
+    std::move(
+        std::istream_iterator<std::string>(in),
+        std::istream_iterator<std::string>(),
+        std::back_inserter(token)
+    );
+
+    if (token.empty())
+        return "";
+
+    std::copy(
+        token.rbegin(), 
+        token.rend(),
+        std::ostream_iterator<std::string>(out, " ")
+    );
+
+    std::string result = out.str();
+
+    return std::move(result.substr(0, result.length() - 1));
```

Comment:
you already returning rvalue, why do you need move here?
and yes, pop_back will be better here